### ISSUE TITLE
docker >= 1.8.3 + marvel plugin renders a host machine network dead

### ISSUE THREAD

**Issue by danielschonfeld** [elastic/elasticsearch#15409]
We've isolated our tests to indicate that when running ES 2.1.0, on a docker 1.8.3+ utilizing kubernetes 1.1.1 a weird networking problem occurs when installing the marvel plugin.  

The problem renders most networking activity dead on the host machine and subsequently on all containers running on that machine, but mostly and 100% it renders the machine unable to resolve any DNS names.

Is marvel dead locking each node internally with a flood of DNS requests to itself?

**Comment by danielschonfeld** [comment]
Update: this issue is not related to kubernetes in any way.  Also not specific to my company image, this problem occurs with the docker hub elasticsearch:2.1.0 image + CoreOS >= 835.9.0.

Steps to reproduce:
1) Need at least 2 machines of CoreOS, 1 should be release >= 835.9.0 and the other a release before that. IE `CoreOS 766.5.0`.

Notes: 
- You will need an ES cluster > 1 node.  2 Nodes is fine as long as they join into a cluster together.
- Machine number 1 will be the one with CoreOS >= 835.9.0
- Machine number 2 will be the one with CoreOS = 766.5.0

2) We happen to use flanneld so for the sake of the experiment please utilize flannel
3) On machine number 1 start ES with the following: `docker run --name es -i -t elasticsearch:2.1.0 /bin/bash`
4) Do the same on machine number 2.
5) On both machines, inside the shell install the license and marvel-agent plugins:
```
/usr/share/elasticsearch/bin/plugin install license
/usr/share/elasticsearch/bin/plugin install marvel-agent
```
6) On machine number 2 edit the `elasticsearch.yaml` and add the discovery line with the IP of machine number 1, i.e.: `discovery.zen.ping.unicast.hosts: [10.244.61.148]`

7) Start ES on machine number 1 `/docker-entry.sh elasticsearch`
8) Do the same on machine number 2
9) Allow both machines to agree they've connected and joined into a cluster

10) Go back to machine number 1 and try to `ping google.com`... You will see the name can't even be resolved. And generally the whole network on the host machine is trashed and so is ES.

**Comment by clintongormley** [comment]
I doubt Marvel is making lots of DNS request (@tlrx could you confirm?)

What about looking at what network sockets are in use?  Also, it seems very weird that this requires two different versions of CoreOs... sounds like it might be an issue with CoreOs then, no?

**Comment by danielschonfeld** [comment]
@clintongormley it requires two versions because in my reproduction steps, I want you to be left with one machine that actually works while the other is trashed.

using an older version archives just that...

**Comment by danielschonfeld** [comment]
@clintongormley also see the linked docker bug... this might not be directly the fault of ES but rather my steps just find a proven way to reproduce a docker bug.  So it might not be ES's fault per say.

**Comment by clintongormley** [comment]
This sounds like a docker problem rather than an ES problem, so I'm going to close this

**Comment by exapsy** [comment]
![image](https://user-images.githubusercontent.com/9019120/98440397-1d963800-2101-11eb-88cc-9f5037e5db9e.png)

This might prove that there's at least a correlation between ES and it having an interaction with host's network. By the time I stopped ES, the network logs stopped. And exactly the time I started it, it showed the logs again. 

This doesn't necessarily mean that ES is causing this though, as I have seriously no idea how the dynamics behind are working either. It might not even be related as far as I know, it just seems interesting that there's a correlation, and it might be helpful.
